Spring Web JAX-B provider NullPointerException

**Describe the bug**
Quarkus Spring Web application with an endpoint producing "application/xml" content fails with a NullPointerException when trying to get JAXB context:
```
[INFO] Running org.acme.ExampleResourceTest
2019-12-04 15:04:12,608 INFO  [io.quarkus] (main) Quarkus 1.0.1.Final started in 0.844s. Listening on: http://0.0.0.0:8081
2019-12-04 15:04:12,611 INFO  [io.quarkus] (main) Profile test activated. 
2019-12-04 15:04:12,612 INFO  [io.quarkus] (main) Installed features: [cdi, resteasy, resteasy-jackson, resteasy-jaxb, spring-di, spring-web]
2019-12-04 15:04:13,716 ERROR [io.qua.ver.htt.run.QuarkusErrorHandler] (vert.x-worker-thread-0) HTTP Request to /hello failed, error id: da24a52f-1c39-406b-a4d2-f0dfabe0c606-1: org.jboss.resteasy.spi.UnhandledException: java.lang.NullPointerException
        at org.jboss.resteasy.core.ExceptionHandler.handleException(ExceptionHandler.java:381)
        at org.jboss.resteasy.core.SynchronousDispatcher.writeException(SynchronousDispatcher.java:209)
        at org.jboss.resteasy.core.SynchronousDispatcher.writeResponse(SynchronousDispatcher.java:587)
        at org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:508)
        at org.jboss.resteasy.core.SynchronousDispatcher.lambda$invoke$4(SynchronousDispatcher.java:252)
        at org.jboss.resteasy.core.SynchronousDispatcher.lambda$preprocess$0(SynchronousDispatcher.java:153)
        at org.jboss.resteasy.core.interception.jaxrs.PreMatchContainerRequestContext.filter(PreMatchContainerRequestContext.java:363)
        at org.jboss.resteasy.core.SynchronousDispatcher.preprocess(SynchronousDispatcher.java:156)
        at org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:238)
        at io.quarkus.resteasy.runtime.standalone.RequestDispatcher.service(RequestDispatcher.java:73)
        at io.quarkus.resteasy.runtime.standalone.VertxRequestHandler.dispatch(VertxRequestHandler.java:118)
        at io.quarkus.resteasy.runtime.standalone.VertxRequestHandler.lambda$handle$0(VertxRequestHandler.java:74)
        at io.vertx.core.impl.ContextImpl.lambda$executeBlocking$2(ContextImpl.java:316)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
        at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.NullPointerException
        at org.jboss.resteasy.plugins.providers.jaxb.AbstractJAXBProvider.findJAXBContext(AbstractJAXBProvider.java:77)
        at org.jboss.resteasy.plugins.providers.jaxb.AbstractJAXBProvider.getMarshaller(AbstractJAXBProvider.java:173)
        at org.jboss.resteasy.plugins.providers.jaxb.AbstractJAXBProvider.writeTo(AbstractJAXBProvider.java:149)
        at org.jboss.resteasy.core.interception.jaxrs.AbstractWriterInterceptorContext.writeTo(AbstractWriterInterceptorContext.java:193)
        at org.jboss.resteasy.core.interception.jaxrs.ServerWriterInterceptorContext.writeTo(ServerWriterInterceptorContext.java:64)
        at org.jboss.resteasy.core.interception.jaxrs.AbstractWriterInterceptorContext.proceed(AbstractWriterInterceptorContext.java:155)
        at org.jboss.resteasy.core.ServerResponseWriter.lambda$writeNomapResponse$2(ServerResponseWriter.java:156)
        at org.jboss.resteasy.core.interception.jaxrs.ContainerResponseContextImpl.filter(ContainerResponseContextImpl.java:404)
        at org.jboss.resteasy.core.ServerResponseWriter.executeFilters(ServerResponseWriter.java:232)
        at org.jboss.resteasy.core.ServerResponseWriter.writeNomapResponse(ServerResponseWriter.java:97)
        at org.jboss.resteasy.core.ServerResponseWriter.writeNomapResponse(ServerResponseWriter.java:70)
        at org.jboss.resteasy.core.SynchronousDispatcher.writeResponse(SynchronousDispatcher.java:578)
        ... 14 more
```


**To Reproduce**
Steps to reproduce the behavior:
1.  git clone https://github.com/gytis/reproducer-quarkus-spring-web-jaxb
2. cd reproducer-quarkus-spring-web-jaxb
3. ./mvnw test

**Environment (please complete the following information):**
 - Output of `uname -a` or `ver`: Darwin gtrikler-mac 19.0.0 Darwin Kernel Version 19.0.0: Thu Oct 17 16:17:15 PDT 2019; root:xnu-6153.41.3~29/RELEASE_X86_64 x86_64
 - Output of `java -version`: OpenJDK 64-Bit GraalVM CE 19.2.1 (build 25.232-b07-jvmci-19.2-b03, mixed mode)
 - GraalVM version (if different from Java): 
 - Quarkus version or git rev: 1.0.1.Final

--- a/extensions/spring-web/deployment/src/main/java/io/quarkus/spring/web/deployment/SpringWebProcessor.java
+++ b/extensions/spring-web/deployment/src/main/java/io/quarkus/spring/web/deployment/SpringWebProcessor.java
@@ -287,6 +287,10 @@ public class SpringWebProcessor {
                     useAllAvailable = true;
                     break OUTER;
                 }
+                if (collectProviders(providersToRegister, categorizedContextResolvers, instance, "produces")) {
+                    useAllAvailable = true;
+                    break OUTER;
+                }
 
                 if (collectProviders(providersToRegister, categorizedReaders, instance, "consumes")) {
                     useAllAvailable = true;

--- a/integration-tests/spring-web/pom.xml
+++ b/integration-tests/spring-web/pom.xml
@@ -25,6 +25,10 @@
         </dependency>
         <dependency>
             <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-resteasy-jaxb</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>io.quarkus</groupId>
             <artifactId>quarkus-spring-di</artifactId>
         </dependency>
         <dependency>

--- a/integration-tests/spring-web/src/main/java/io/quarkus/it/spring/web/Book.java
+++ b/integration-tests/spring-web/src/main/java/io/quarkus/it/spring/web/Book.java
@@ -1,0 +1,24 @@
+package io.quarkus.it.spring.web;
+
+import javax.xml.bind.annotation.XmlRootElement;
+
+@XmlRootElement
+public class Book {
+
+    private String name;
+
+    public Book() {
+    }
+
+    public Book(String name) {
+        this.name = name;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public void setName(String name) {
+        this.name = name;
+    }
+}

--- a/integration-tests/spring-web/src/main/java/io/quarkus/it/spring/web/BookController.java
+++ b/integration-tests/spring-web/src/main/java/io/quarkus/it/spring/web/BookController.java
@@ -1,0 +1,14 @@
+package io.quarkus.it.spring.web;
+
+import org.springframework.http.MediaType;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+public class BookController {
+
+    @GetMapping(produces = MediaType.APPLICATION_XML_VALUE, path = "/book")
+    public Book someBook() {
+        return new Book("Guns germs and steel");
+    }
+}

--- a/integration-tests/spring-web/src/test/java/io/quarkus/it/spring/web/SpringControllerTest.java
+++ b/integration-tests/spring-web/src/test/java/io/quarkus/it/spring/web/SpringControllerTest.java
@@ -204,4 +204,13 @@ public class SpringControllerTest {
                 .body(containsString("hello from error"))
                 .statusCode(402);
     }
+
+    @Test
+    public void testMethodReturningXmlContent() {
+        RestAssured.when().get("/book")
+                .then()
+                .statusCode(200)
+                .contentType("application/xml")
+                .body(containsString("steel"));
+    }
 }

